app/maptools/recordingmaptool.cpp

```diff
@@ -412,9 +420,43 @@ void RecordingMapTool::cancelGrab()
   }
 
   setState( MapToolState::View );
+  toggleSelectedVertexVisibility( -1 );
   setActiveVertex( Vertex() );
 }
 
+void RecordingMapTool::toggleSelectedVertexVisibility( int vertexIndex )
+{
+  if ( !mExistingVertices.isEmpty() )
+  {
+    // add currently selected vertex back to the mExistingVertices in order
+    // to make it visible again
+    if ( mActiveVertex.isValid() )
+    {
+      QgsMultiPoint *points = qgsgeometry_cast<QgsMultiPoint *>( mExistingVertices.get() );
+      points->addGeometry( mActiveVertex.coordinates().clone() );
+    }
+
+    if ( vertexIndex != -1 )
+    {
+      mActiveVertex = mVertices.at( vertexIndex );
+
+      // remove newly selected vertex from the mExistingVertices
+      QgsPoint p = mActiveVertex.coordinates();
+      mExistingVertices.filterVertices( [p]( const QgsPoint & point )->bool
+      {
+        return !InputUtils::equals( point, p, 1e-16 );
+      } );
+    }
+    emit existingVerticesChanged( mExistingVertices );
+  }
+}
+
+double RecordingMapTool::pixelsToMapUnits( double numPixels )
+{
+  QgsRenderContext context = QgsRenderContext::fromMapSettings( mapSettings()->mapSettings() );
+  return numPixels * context.scaleFactor() * context.mapToPixel().mapUnitsPerPixel();
```

Comment:
I believe we will need to incorporate DPI here as well

Reply:
Can you please multiply the number of pixels by `InputUtils::calculateScreenDpr`?